packages/vitest/src/types/mocker.ts

```diff
@@ -1,6 +1,9 @@
 export type MockFactoryWithHelper<M = unknown> = (
-  importOriginal: <T extends M>() => Promise<T>
-) => any
+  importOriginal: () => Promise<M>
+) => Partial<Record<keyof M, any>>
+export type PromiseMockFactoryWithHelper<M = unknown> = (
+  importOriginal: () => Promise<M>
+) => Promise<Partial<Record<keyof M, any>>>
```

Comment:
1. Why is it partial?
2. Why do you need a separate `Promise` one and won't just do `| Promise<T>`?

Reply:
1. I don't think everyone wants to mock all the exports.
2. The union type like `| Promise<T>` make wrong type like below screenshot. 
<img width="475" alt="스크린샷 2024-07-16 오전 12 48 54" src="https://github.com/user-attachments/assets/3a2ab9b1-f967-4eb4-b94d-8c28f3a80025">

Reply:
> I don't think everyone wants to mock all the exports.

If they want typechecking, they need to mock it fully. We already throw an error if something is not returned.

Reply:
> If they want typechecking, they need to mock it fully. We already throw an error if something is not returned.

hmm... I think so. Because if you want to implement partially, they can call importOriginal.

> Seems fine to me

That is not async function but, typescript provides promise method.

Reply:
> That is not async function

You can return a `new Promise` from every function - in fact any object with `then` method can be awaited, so TypeScript is not wrong here

Reply:
I think I was wrong, but I think it might not be normal to be offered auto-completion for promise method, so I'm submitting PR in typescript.
But if it's okay, I think we can revise it as you suggested.